src/http/http_response.f90

```diff
@@ -36,6 +39,24 @@ subroutine append_header(this, key, value)
         end if
     
     end subroutine append_header
-  
 
+    pure function header_value(this, key) result(retval)
+        class(response_type), intent(in) :: this
+        character(*), intent(in) :: key
+        character(:), allocatable :: retval
+        type(string_type) :: string_to_match
+        integer :: i
+        
+        string_to_match = to_lower(string_type(key))
+        
+        retval = ''
+
```

Comment:
On second thought, I suggest returning an unallocated character string as a result if the key is not found. An unallocated string is more easily caught as erroneous.

```suggestion
```